lib/Raisin/Middleware/Formatter.pm

```diff
@@ -36,13 +36,16 @@ sub call {
         $env->{'raisinx.body_params'} = $d->deserialize($req->content);
     }
 
-    my $format = $self->negotiate_format($req);
+    my $res = $self->app->($env);
+    my $content_format  = Plack::Util::header_get($res->[1], 'Content-type');
+    my $format = $content_format || $self->negotiate_format($req);
```

Comment:
Could you please elaborate what are you going to achieve here?

Reply:
Let us say we have an endpoint that sends out a pdf we need to add header as application/pdf. If we set api_default_format  then this is set for all over the app. In our case we want specific API endpoint to send different headers. One option is this can be done via Accept headers which is not in our control since we have existing endpoints that works well without specifying this. 

For example in the existing sample-app if we were to add an endpoint that sends out pdf then there seems to be no way to set this from the app side. we will have to then have the client send us header to get it working.

Let me know if you have questions or other suggestions on how to do this

Reply:
In other words you want content-type you set inside of endpoint had precedence, right?

But if you're sending `Accept: application/json` and get back `Content-Type: application/pdf`, is it correct? Shouldn't you receive `406 Not Acceptable` then?

So the problem here is that you want to set formatters per endpoint so they have precedence over the global one. If so, that should be done in a different way then.